### PR TITLE
[improvement]: move vpc interface to eth0

### PR DESCRIPTION
**Pull request by rahulait**
**What this PR does / why we need it**:
This PR moves vpc interface from eth1 to eth0. This simplifies networking for clusters provisioned within VPC.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/flavors/kubeadm/default/kustomization.yaml [hunk 1]
```diff
@@ -4,7 +4,6 @@ resources:
   - ../../../infra
   - kubeadmConfigTemplate.yaml
   - kubeadmControlPlane.yaml
-  - allow-etcd-policy.yaml
```
Comment (eljohnson92) [inline templates/flavors/kubeadm/default/kustomization.yaml:7]: I think we have some docs noting that this is being set, could you also update to reflect that this is no longer added through a CCNP in our docs?

**Comment by AshleyDumaine** [comment]
LGTM aside from doc change needed on docs/src/topics/vpc.md 👍

File: templates/flavors/kubeadm/default/kustomization.yaml (hunk 1 above)
Comment (rahulait) [reply templates/flavors/kubeadm/default/kustomization.yaml:7]: Updated it. We do require this for kubeadm vpcless flavors as etcd there still uses public ip. Updated docs to reflect the same.

File: templates/flavors/kubeadm/vpcless/kustomization.yaml [hunk 2]
```diff
@@ -63,14 +64,3 @@ patches:
     patch: |-
       - op: remove
         path: /spec/kubeadmConfigSpec/initConfiguration/skipPhases
-  - target:
```
Comment (AshleyDumaine) [inline templates/flavors/kubeadm/vpcless/kustomization.yaml:66]: do we also want to remove this from the rke2 and k3s flavors for consistency?

Comment (rahulait) [reply templates/flavors/kubeadm/vpcless/kustomization.yaml:66]: updated.

**Review by AshleyDumaine — APPROVED** [review]
LGTM

**Pull request merged**